### PR TITLE
improve our handling of situations where we don't have full env variables

### PR DESCRIPTION
**Pull request by devoncarew**
Improve our handling of situations where we don't have full env variables. Mac only:
- detect the current shell and the user's preferred shell
- gather all the env variables from the preferred shell
- when exec'ing a process, pass in the env variables discovered from the user's shell

@danrubel

### CODE REVIEW

File: lib/node/process.dart [hunk 1]
```diff
@@ -180,3 +176,57 @@ class ProcessResult {
...
+  try {
+    return execSync('echo \$$variable');
+  } catch (e) {
+    return null;
+  }
+}
+
+/// This class exists to help manage situations where Atom is running in an
+/// environment without a properly set up environment (env and PATH variables).
+class MacShellWrangler {
```
Comment (danrubel) [inline lib/node/process.dart:201]: A little more here would be helpful. /bin/sh does not work but /bin/bash does? Problem exists only on Macs? Just need a clear description of the problem encountered and the solution implemented.

**Comment by danrubel** [comment]
LGTM

File: lib/node/process.dart (hunk 1 above)
Comment (danrubel) [inline lib/node/process.dart:201]: Maybe just add you PR comment here.

Comment (devoncarew) [inline lib/node/process.dart:201]: Will do! It's taken a lot of noodling to get here. Would be good to capture that into in the comments.

**Pull request merged**